README.md

```diff
@@ -15,55 +15,279 @@ Creates a Docker container with an installation of the
 server ([dovecot](https://dovecot.org)) for accessing the archvies
 of sent email.  All email is BCC'd to the `mailarchive` account.
 
-## Usage ##
+## Running ##
 
-### Install ###
+### Running with Docker ###
 
-Pull `cisagov/postfix` from the Docker repository:
+To run the `cisagov/postfix` image via Docker:
 
-    docker pull cisagov/postfix
+```console
+docker run cisagov/postfix:0.0.4
+```
 
-Or build `cisagov/postfix` from source:
+### Running with Docker Compose ###
 
-    git clone https://github.com/cisagov/postfix-docker.git
-    cd postfix-docker
-    docker-compose build --build-arg VERSION=0.0.1
+1. Create a `docker-compose.yml` file similar to the one below to use [Docker Compose](https://docs.docker.com/compose/)
+or use the [sample `docker-compose.yml`](docker-compose.yml) provided with
+this repository.
+
+    ```yaml
+    ---
+    version: "3.7"
+
+    services:
+      postfix:
+        build:
+          # VERSION must be specified on the command line:
+          # e.g., --build-arg VERSION=0.0.4
+          context: .
+          dockerfile: Dockerfile
+        image: cisagov/postfix
+        init: true
+        restart: always
+        environment:
+          - PRIMARY_DOMAIN=example.com
+          - RELAY_IP=172.16.202.1/32
+        networks:
+          front:
+            ipv4_address: 172.16.202.2
+        ports:
+          - target: "25"
+            published: "1025"
+            protocol: tcp
+            mode: host
+          - target: "587"
+            published: "1587"
+            protocol: tcp
+            mode: host
+          - target: "993"
+            published: "1993"
+            protocol: tcp
+            mode: host
+
+    networks:
+      front:
+        driver: bridge
+        ipam:
+          driver: default
+          config:
+            - subnet: 172.16.202.0/24
+    ```
+
+1. Start the container and detach:
+
+    ```console
+    docker-compose up --detach
+    ```
+
+## Using secrets with your container ##
+
+This container also supports passing sensitive values via [Docker
+secrets](https://docs.docker.com/engine/swarm/secrets/).  Passing sensitive
+values like your credentials can be more secure using secrets than using
+environment variables.  See the
+[secrets](#secrets) section below for a table of all supported secret files.
+
+1. To use secrets, populate the following files in the `src/secrets` directory:
+
+- `fullchain.pem`
+- `privkey.pem`
+- `users.txt`
```

Comment:
Unrelated to this PR, but we don't have a `.gitignore` entry to prevent something in `src/secrets` from being added accidentally. We should probably get an issue in to fix that if the README is going to suggest putting actual secrets in the directory.

Reply:
> Unrelated to this PR, but we don't have a `.gitignore` entry to prevent something in `src/secrets` from being added accidentally. We should probably get an issue in to fix that if the README is going to suggest putting actual secrets in the directory.

I suppose we could rename `src/secrets` to something like `src/example_secrets` and then create a new, empty `src/secrets` and some explanatory text in the README.  Thoughts from the rest of @cisagov/team-ois?

Reply:
This seems like a reasonable enough approach to me.

Reply:
@jsf9k - What sayeth thou?

Reply:
This is at least the second time you guys have brought up this concern about someone accidentally committing real secrets in `src/secrets`, so we should probably look at fixing the issue in [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker).  Isn't that directory only present and non-empty in order to give the example `docker-compose.yml` file from the skeleton fake data to use when the tests are run?  If that is the case, then I'd prefer we did `git mv -r src/secrets tests && git mv docker-compose.yml tests` in [cisagov/skeleton-docker](https://github.com/cisagov/skeleton-docker) and correct references in those files and others as necessary.  I think that's where that testing-specific jazz belongs.  The `README.md` can point there to highlight an example composition and secrets files.

TL;DR:  I don't think of Docker secrets as part of the source of a Docker image.

N.B.:  You can't really create an empty directory in `git`.  If there isn't any content there then that directory won't be included when you `git clone`.

Reply:
I created https://github.com/cisagov/skeleton-docker/issues/68 to track this issue.  @mcdonnnj - is that sufficient for now to resolve this conversation?